File: adminpanel.Web/Modules/ServerTypes/ecommercecom/categoryForm.ts
import { StringEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface categoryForm {
    Name: StringEditor;
    Description: StringEditor;
    Photo: StringEditor;
}

export class categoryForm extends PrefixedContext {
    static readonly formKey = 'ecommercecom.category';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!categoryForm.init)  {
            categoryForm.init = true;

            var w0 = StringEditor;

            initFormType(categoryForm, [
                'Name', w0,
                'Description', w0,
                'Photo', w0
            ]);
        }
    }
}

File: adminpanel.Web/Modules/ecommercecom/category/categoryForm.cs
using Serenity.ComponentModel;

namespace adminpanel.ecommercecom.Forms;

[FormScript("ecommercecom.category")]
[BasedOnRow(typeof(categoryRow), CheckNames = true)]
public class categoryForm
{
    public string Name { get; set; }
    public string Description { get; set; }
    public string Photo { get; set; }
}